runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -373,10 +350,39 @@ private StreamingDataflowWorker(
             clock,
             options.getActiveWorkRefreshPeriodMillis(),
             stuckCommitDurationMillis,
-            () -> Collections.unmodifiableCollection(computationMap.values()),
+            computationStateCache::getAllComputations,
             sampler,
             metricTrackingWindmillServer::refreshActiveWork,
             executorSupplier.apply("RefreshWork"));
+
+    WorkerStatusPages workerStatusPages =
+        WorkerStatusPages.create(DEFAULT_STATUS_PORT, memoryMonitor, () -> true);
+    this.statusPages =
+        windmillServiceEnabled
+            ? StreamingWorkerStatusPages.forStreamingEngine(
```

Comment:
would a builder pattern be better than the separate forStreamingEngine and forAppliance factory methods?

There seem to be a lot of common things.

Reply:
done